internal/evaluator/conftest_evaluator.go

```diff
@@ -872,15 +872,46 @@ func strictCapabilities(ctx context.Context) (string, error) {
 	return string(blob), nil
 }
 
-func computeIncludeExclude(sc *ecc.SourceConfig, p policy.Policy) ([]string, []string) {
+func computeIncludeExclude(src ecc.Source, p policy.Policy) ([]string, []string) {
 	var include, exclude []string
 
+	sc := src.Config
+
 	// The lines below take care to make a copy of the includes/excludes slices in order
 	// to ensure mutations are not unexpectedly propagated.
 	if sc != nil && (len(sc.Include) != 0 || len(sc.Exclude) != 0) {
 		include = append(include, sc.Include...)
 		exclude = append(exclude, sc.Exclude...)
-	} else if policyConfig := p.Spec().Configuration; policyConfig != nil {
+	}
+
+	vc := src.VolatileConfig
+	if vc != nil {
+		at := p.EffectiveTime()
+		filter := func(items []string, criteria []ecc.VolatileCriteria) []string {
+			for _, c := range criteria {
+				from, err := time.Parse(time.RFC3339, c.EffectiveOn)
+				if err != nil {
+					log.Warnf("unable to parse time for criteria %q, was given %q: %v", c.Value, c.EffectiveOn, err)
+					from = at.Add(-1)
+				}
+				until, err := time.Parse(time.RFC3339, c.EffectiveUntil)
+				if err != nil {
+					log.Warnf("unable to parse time for criteria %q, was given %q: %v", c.Value, c.EffectiveUntil, err)
+					until = at.Add(1)
+				}
+				if from.Before(at) && until.After(at) {
```

Comment:
I wonder if this should be `!from.After(at) && !until.Before(at)` in order to create an inclusive range which seems less surprising, I think.

It would also allow both `from` and `until` to have the default value of `at`.

Reply:
Yeah, I think I want to use Compare instead, makes things more explicit